### PR TITLE
add docs for lua filter and change log

### PR DESCRIPTION
**Pull request by fl0Lec**
followup of https://github.com/envoyproxy/envoy/pull/36920

Commit Message: add docs for lua filter and change log
Additional Description:
Risk Level: none
Testing: no
Docs Changes: yes
Release Notes: yes
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@fl0Lec docs failure is real

```console
 Warning: /tmp/tmp7b4f_pq0/generated/rst/version_history/v1.33/v1.33.0.rst:144: WARNING: undefined label: 'config_http_filters_lua_stream_info_downstream_direct_local_address'
```

File: docs/root/configuration/http/http_filters/lua_filter.rst [hunk 1]
```diff
@@ -794,6 +794,16 @@ protocol()
 Returns the string representation of :repo:`HTTP protocol <envoy/http/protocol.h>`
 used by the current request. The possible values are: ``HTTP/1.0``, ``HTTP/1.1``, ``HTTP/2`` and ``HTTP/3*``.
 
+downstreamDirectLocalAddress()
```
Comment (kyessenov) [inline docs/root/configuration/http/http_filters/lua_filter.rst:797]: You need to put the anchor here for the link to work.

**Comment by KBaichoo** [comment]
/docs

should render the docs for you to validate

**Comment by KBaichoo** [comment]
/wait

For addressing comment above re linking

File: docs/root/configuration/http/http_filters/lua_filter.rst [hunk 2]
```diff
@@ -794,6 +794,18 @@ protocol()
 Returns the string representation of :repo:`HTTP protocol <envoy/http/protocol.h>`
 used by the current request. The possible values are: ``HTTP/1.0``, ``HTTP/1.1``, ``HTTP/2`` and ``HTTP/3*``.
 
+.. _config_http_filters_lua_stream_info_downstream_direct_local_address:
+
+downstreamDirectLocalAddress()
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/lua_filter.rst:799]:
```suggestion
``downstreamDirectLocalAddress()``
```

File: docs/root/configuration/http/http_filters/lua_filter.rst [hunk 3]
```diff
@@ -794,6 +794,18 @@ protocol()
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: lua
+
+  streamInfo:downstreamDirectLocalAddress()
+
+Returns the string representation of :repo:`downstream direct local address <envoy/stream_info/stream_info.h>`
+used by the current request. This is always the physical local address of the connection.
+
 downstreamLocalAddress()
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/lua_filter.rst:809]:
```suggestion
``downstreamLocalAddress()``
```

File: docs/root/configuration/http/http_filters/lua_filter.rst (hunk 2 above)
Comment (fl0Lec) [reply docs/root/configuration/http/http_filters/lua_filter.rst:799]: I am not sure I understand this change, would you like me to do it for the whole file ?
I does not seems to be the case for the other method

Comment (phlax) [reply docs/root/configuration/http/http_filters/lua_filter.rst:799]: this just makes it a `literal` - ie its not an english word - its literally this

anywhere this is used in docs it should be made literal - with the exception of in link text as rst/sphinx does not currently handle literal links

Comment (fl0Lec) [reply docs/root/configuration/http/http_filters/lua_filter.rst]: make sense, I replace everywhere where applicable in the file

Comment (phlax) [reply docs/root/configuration/http/http_filters/lua_filter.rst]: great, thanks

**Comment by phlax** [comment]
thanks for updating the other headings @fl0Lec 

looks like one was missed tho

image

**Review by phlax — APPROVED** [review]
lgtm, thanks @fl0Lec

**Pull request merged**